File: abc365b/abc365b.ml
open Printf
open Scanf

let solve n l =
  snd (List.nth (List.sort compare l) (n - 2))

let () =
  scanf "%d " (fun n ->
    List.init n (fun i -> scanf "%d " (fun x -> (x, i + 1))) |> solve n) |> printf "%d\n"

File: abc365b/abc365b.hs
{-# LANGUAGE OverloadedStrings #-}
import Data.Text (Text)
import qualified Data.Text as T
import qualified Data.Text.IO as T
import Data.Text.Read (Reader)
import qualified Data.Text.Read as T
import Data.List (sortBy)

main :: IO ()
main = getLine >> solve <$> f >>= print
  where
    f = map (readtx T.decimal) <$> T.words <$> T.getLine

solve :: [Int] -> Int
solve = snd . head . tail . sortBy (flip compare) . flip zip [1..]

readtx :: Num a => Reader a -> Text -> a
readtx f = either (const 0) fst . f
